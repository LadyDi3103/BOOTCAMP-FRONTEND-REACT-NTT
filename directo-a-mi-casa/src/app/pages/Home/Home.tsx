import "../Home/Home.css";
import CallToAction from "../../../shared/components/CallToAction/CallToAction";
import Carousel from "../../../components/Carousel/Carousel";
import CategoriesSection from "../../../components/CategorySection/CategoriesSection";
import CategoryItem from "../../../components/CategoryItem/CategoryItem";
import SpecialOffers from "../../../components/SpecialOffers/SpecialOffers";

const Home = () => {
    return (
        <main>

            <CallToAction />

            <Carousel />

            <CategoriesSection />

            <CategoryItem />

            <SpecialOffers />
        </main>
    );
};

export default Home;
