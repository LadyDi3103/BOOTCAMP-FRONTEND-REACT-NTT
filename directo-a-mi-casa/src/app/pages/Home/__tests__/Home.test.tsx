import { render, screen } from "@testing-library/react";
import Home from "../Home";

jest.mock("@/shared/components/CallToAction/CallToAction", () => () => <div data-testid="call-to-action">CallToAction</div>);
jest.mock("@/components/Carousel/Carousel", () => () => <div data-testid="carousel">Carousel</div>);
jest.mock("@/shared/components/CategorySection/CategoriesSection", () => () => <div data-testid="categories-section">CategoriesSection</div>);
jest.mock("@/shared/components/CategoryItem/CategoryList", () => () => <div data-testid="categories">CategoriesList</div>);

describe("Home Page", () => {
    test("Should render all components correctly", () => {
        render(<Home />);

        expect(screen.getByTestId("call-to-action")).toBeInTheDocument();
        expect(screen.getByTestId("carousel")).toBeInTheDocument();
        expect(screen.getByTestId("categories-section")).toBeInTheDocument();
        expect(screen.getByTestId("categories")).toBeInTheDocument();
    });

    test("Should check the content of the components", () => {
        render(<Home />);

        expect(screen.getByTestId("call-to-action")).toHaveTextContent("CallToAction");
        expect(screen.getByTestId("carousel")).toHaveTextContent("Carousel");
        expect(screen.getByTestId("categories-section")).toHaveTextContent("CategoriesSection");
        expect(screen.getByTestId("categories")).toHaveTextContent("CategoriesList");
    });

    test("Should match snapshot", () => {
        const { container } = render(<Home />);
        expect(container).toMatchSnapshot();
    });
});