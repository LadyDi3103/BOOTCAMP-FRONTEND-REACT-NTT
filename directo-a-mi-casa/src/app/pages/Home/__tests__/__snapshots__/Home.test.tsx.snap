// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should match snapshot 1`] = `
<div>
  <main>
    <div
      data-testid="call-to-action"
    >
      CallToAction
    </div>
    <div
      data-testid="carousel"
    >
      Carousel
    </div>
    <div
      data-testid="categories"
    >
      CategoriesList
    </div>
    <div
      data-testid="categories-section"
    >
      CategoriesSection
    </div>
  </main>
</div>
`;
